framework/web/Controller.php

```diff
@@ -50,6 +50,32 @@ public function renderAjax($view, $params = [])
     }
 
     /**
+     * Send unformatted data.
+     *
+     * This method is a shortcut for sending unformatted data. It will return
+     * the [[Application::getResponse()|response]] application component after configuring
+     * the [[Response::$format|format]] and setting the [[Response::$data|data]].
+     * A common usage will be:
+     *
+     * ```php
+     * return $this->asRaw($data);
+     * ```
+     *
+     * @param mixed $data the data that should be sent.
+     * @return Response a response that is configured to send unformatted `$data`.
+     * @since 2.0.12
+     * @see Response::$format
+     * @see Response::FORMAT_RAW
+     * @see ResponseFormatterInterface
+     */
+    public function asRaw($data)
+    {
+        $format = Response::FORMAT_RAW;
+        $response = $this->formatData($format, $data);
+        return $response;
+    }
```

Comment:
> It's a well known good practice not to make any calculations in return statements.

Do you suggest that using 
```php
return $this->asJson($data);
```
in controller is a bad practice? :D

Using variable make sense only for complicated calculations, here you have only one method call with pretty obvious logic.